runtime/ftplugin/help.vim

```diff
@@ -18,5 +18,45 @@ if has("conceal")
   setlocal cole=2 cocu=nc
 endif
 
+function! s:create_toc() abort
+  if !exists('b:help_toc')
+    let b:help_toc = []
+    let lnum = 1
+    let last_line = line('$')
+    let last_added = 0
+
+    while lnum <= last_line
+      let add_text = ''
+      let text = getline(lnum)
+
+      if text =~# '^=\+$'
+        let text = getline(lnum + 1)
+        if text =~# '\*[^*]\+\*'
+          let lnum += 1
+          let add_text = matchstr(text, '.\{-}\ze\s*\*')
+        endif
+      elseif text =~# '^[A-Z][-A-Z .][-A-Z0-9 .()]*[ \t]\+\*'
```

Comment:
Because some lines start with non-[A-Z] characters and I felt the ones that I saw weren't good TOC entries.